specs/protocol/identifiers.md

```diff
@@ -24,14 +23,9 @@ For a transaction of type `TransactionType.Create`, `tx`, the contract ID is `sh
 
 Is represented as an _outpoint_: a pair of [transaction ID](#transaction-id) as `byte[32]` and output index as a `uint8`.
 
+### Message ID
 
-### Deposit ID
-
-The ID of a deposit is computed as the [hash](./cryptographic_primitives.md#hashing) of the [Deposit Asset ID](#deposit-asset-id) `byte[32]`, recipient address `byte[32]`, amount `uint64` and deposit nonce `uint64`: `hash(byte[32] ++ byte[32] ++ uint64 ++ uint64)`. All value types are serialized according to the standard [transaction serialization](./tx_format.md#transaction).
-
-#### Deposit Asset ID
-
-The ID of a deposit asset is computed as the [hash](./cryptographic_primitives.md#hashing) of the token address `byte[32]` and token precision `uint8`: `hash(byte[32] ++ uint8)`. The address value is serialized as a byte array of length 32 left-padded with zeroes, and all other value types are serialized according to the standard [transaction serialization](./tx_format.md#transaction).
+The ID of a message is computed as the [hash](./cryptographic_primitives.md#hashing) of the sender address `byte[32]`, recipient address `byte[32]`, the message data (either [EVM abi](https://docs.soliditylang.org/en/v0.8.13/abi-spec.html) or [FuelVM abi](./abi.md)/data) `byte[]`, the amount being sent with the message `uint64`, the message nonce `uint64` and the message owner (either address or predicate hash) `byte[32]`: `hash(byte[32] ++ byte[32] ++ byte[] ++ uint64 ++ uint64 ++ byte[32])`. The address values are serialized as a byte array of length 32 left-padded with zeroes, and all other value types are serialized according to the standard [transaction serialization](./tx_format.md#transaction). If the owner component is not applicable then a 32 byte long array of zeros should be used.
```

Comment:
There should be a data length too, no?
```suggestion
The ID of a message is computed as the [hash](./cryptographic_primitives.md#hashing) of:

1. the sender address `byte[32]`,
1. recipient address `byte[32]`,
1. the message data `byte[]`,
1. the amount being sent with the message `uint64`,
1. the message nonce `uint64`,
1. the message owner (either address or predicate hash)

`byte[32]`: `hash(byte[32] ++ byte[32] ++ byte[] ++ uint64 ++ uint64 ++ byte[32])`. The address values are serialized as a byte array of length 32 left-padded with zeroes, and all other value types are serialized according to the standard [transaction serialization](./tx_format.md#transaction). If the owner component is not applicable then a 32 byte long array of zeros should be used.
```

Reply:
Not necessarily. In the solidity contracts I haven't been using an explicitly defined length. It technically reduces gas cost for hashing, but not by a lot if we find it extra useful somewhere else

Reply:
We probably want the data length otherwise there's no way to know how long it is. I guess there's only one variable-length field, so you could infer it solely by looking at the total date length, but that seems kind of brittle.